### PR TITLE
Add authentication and authorization

### PR DESCRIPTION
**Pull request by tiran**
- parse 'x-rh-identity' header in authentication middleware
- set auth credential scope based on header presence and values

The authentication middle returns AuthCredentials and a user object. The user object contains information about 'who' performed the request. AuthCredentials define the scope for authorization. In our case the user object also has the org id, type (`User` or `System`), and additional information. For requests with a correct `x-rh-identity` header the scope is `authenticated` and either `User` or `System`.

Starlette's authentication API mandates that a route function either accepts a `request` argument for HTTP routes or a `websocket` argument for WebSocket routes. Other argument names are not supported.

### CODE REVIEW

**Comment by tiran** [comment]
The 3scale container now adds a `x-rh-identity` header. Requests with TLS client cert auth get a `System` header, other requests a `User` header.

File: webconsoleapp-local.yaml [hunk 1]
```diff
@@ -32,6 +32,8 @@ spec:
           value: https://localhost:{PORT_3SCALE}
         - name: SESSION_INSTANCE_DOMAIN
           value: .dns.podman
+        - name: FAKE_AUTHENTICATION
```
Comment (cryptomilk) [inline webconsoleapp-local.yaml]: You're removing this in an earlier patch and adding it back here. Do not remove it in the first place?

Comment (tiran) [reply webconsoleapp-local.yaml]: GitHub UI does not play well with force pushes. We'll squash the history when the PR is ready.

File: appservice/multiplexer.py [hunk 2]
```diff
@@ -482,6 +472,27 @@ async def update_session(session_id, status):
     await REDIS.publish('sessions', dumped_sessions)
 
 
+def get_session(conn: HTTPConnection) -> Tuple[str, Dict[str, Union[str, int]]]:
+    """Get session from request/websocket
+
+    Raises 404 for unknwon session ids
```
Comment (jelly) [inline appservice/multiplexer.py:478]:
```suggestion
    Raises 404 for unknown session ids
```

**Comment by jelly** [comment]
Testing this now on test.cloud.redhat.com: (TL;DR it works)

Everything seems to work fine, registering a temporary new VM (rhel-9-2) with subscription-manager and getting a certificate and key in /etc/pki/consumer.

Using:
```
/tmp/cockpit-bridge-websocket-connector.pyz --tls-cert /etc/pki/consumer/cert.pem --tls-key /etc/pki/consumer/key.pem -k wss://test.cloud.redhat.com/wss/webconsole/v1/sessions/925b30dd-0ec3-4175-84a7-187bd97db365/ws
```

https://test.cloud.redhat.com/wss/webconsole/v1/sessions/925b30dd-0ec3-4175-84a7-187bd97db365/web/ 

Correctly redirects but then receives a error with a JSON response:
```
{"errors":[{"meta":{"response_by":"gateway"},"detail":"Insights services authentication failed","status":401}]}
```

I should have used `-d` with `/tmp/cockpit-bridge-websocket-connector.pyz`, but it turns out the Virtual machine wasn't properly registered maybe the webconsole should also debug log authentication failures?

**Review by jelly — APPROVED** [review]
Thanks!

**Pull request merged**